controllers/shooter.py

```diff
@@ -0,0 +1,90 @@
+from enum import Enum
+
+from magicbot import StateMachine, state
+
+from components.intake import Intake
+from components.shooter import Shooter
+from components.tilt import Tilt
+from components.turret import Turret
+
+
+class GoalHeight(Enum):
+    HIGH = 2
+    MID = 1
+    LOW = 0
+
+
+class ShooterController(StateMachine):
+    intake_component: Intake
+    shooter_component: Shooter
+    tilt: Tilt
+    turret: Turret
+
+    INTAKE_AZIMUTH: float
+    INTAKE_TILT: float
+
+    def __init__(self) -> None:
+        # set default preference for goal height
+        pass
+
+    @state(must_finish=True)
+    def preparing_intake(self) -> None:
+        # send pitch and turret to angle
+
+        # if at pitch and turret angle
+        # next state intaking
+        pass
+
+    @state(must_finish=True)
+    def intaking(self) -> None:
+        # shooter intake
+        # intake deploy
+
+        # if shooter is loaded
+        # retract intake
+        # next state tracking
+        pass
+
+    # does this need to be must finish so it always runs or should we call engage when we want this state to run
+    @state(first=True)
+    def tracking(self) -> None:
```

Comment:
Sounds like we want a default state. I'm not sure what's being described here is actually the desirable behaviour though; we shouldn't track a node if we don't have a game piece to shoot!